app/views/rooms/show.html.erb

```diff
@@ -1,4 +1,22 @@
-<h1>Rooms#show</h1>
-<p>Find me in app/views/rooms/show.html.erb</p>
-<!-- TODO: Display rooms data: Name, Location, Picture(s), Price per night -->
-<!-- TODO: Allow to reserve the room for various periods of time -->
+<div class="container">
+  <h1><%= @room.name %></h1>
+
+  <p><strong>Address:</strong> <%= @room.address %></p>
+  <p><strong>Images:</strong></p>
+  <div class="row">
+    <% @room.images.each do |image| %>
+      <div class="col-md-4">
+        <%= image_tag url_for(image), class: "img-thumbnail" %>
+      </div>
+    <% end %>
+  </div>
+  <p><strong>Rent per Night:</strong> <%= humanized_money_with_symbol(@room.price_per_night) %></p>
+
+  <br>
+
+  <% if @room.user_id == current_user&.id %>
+    <%= link_to 'Edit', edit_my_room_path(@room), class: 'btn btn-secondary' %> |
+  <% end %>
```

Comment:
My bad, I did not see it there. I was looking for it in `app/views/rooms/show.html.erb` since the TODO is located in this file. Glad I did not reject your test for this. :)